internal/distribution/compatibility.go

```diff
@@ -26,6 +27,28 @@ type VersionRange struct {
 	Max string
 }
 
+// GetConfigKinds returns the list of supported configuration kinds.
+func GetConfigKinds() []string {
+	// We should get this list from the supported APIs instead of hardcoding it,
+	// but AFAIK we don't have a way to do that yet.
+	return []string{
+		EKSClusterKind,
+		KFDDistributionKind,
+		OnPremisesKind,
+	}
+}
+
+// ValidateConfigKind checks if the given kind is supported and returns the normalised value for the Kind and an error.
+func ValidateConfigKind(kind string) (string, error) {
```

Comment:
Maybe move the validation functions inside a new file named `internal/distribution/kind.go`? Or something like that, to separate it from the `compatibility.go`. I am super open to hear your opinions on this.

Reply:
I did NOT want to do a major refactor with this feature, this PR should have been even simpler. But I took the chance to centralised at least some definitions.

I'm fine with moving them to a separated file. I evaluated several places but all felt a little odd or caused circular import issues that would have meant to do more refactor to fix.

Ideally, I think that this (the available Kinds and related funcs) should be somewhere in the distribution's API module instead or the equivalent to it in this repo (that's where the configuration kinds come from), but again doing that would have ended up in a scope creep of this PR.

Overall the module structure of furyctl (and the distribution's go modules) is not clear to me.

Reply:
Yeah, I agree that we need to clarify the whole architecture a bit more and maybe have some central design documents available as a reference.